src/model/transactions.coffee

```diff
@@ -59,3 +61,14 @@ TransactionSchema.index "request.timestamp"
 
 #compile schema into Model
 exports.Transaction = connectionDefault.model 'Transaction', TransactionSchema
+
+exports.Transaction.collection.createIndex({
+  'request.timestamp': 1
+  channelID: 1
+},
+{
+  name: 'transaction.autoRetry'
```

Comment:
The partial index is built on the expression, so it will only include transactions where internalServerError is true (and wasRerun false). It basically a handle on all those transactions that need to be rerun; so I don't think it would be useful in a generic way.